docs_src/design/ucr/0005-Hybrid-App-Device-Services.md

```diff
@@ -0,0 +1,179 @@
+- New section in edgex-docs describing how the Hybrid App-Device Service approach fits into the EdgeX framework
+
+### Discussion and Example Hybrid App-Device Service Descriptions
+
+To illustrate the use of this hybrid solution, consider the needs common to five representative 
+Application Services as the single Use Case in this proposal: 
+two analytic (Trending and Alarming), one utility (NTP client), and two north-bound (BACnet/IP and Cloud Service). 
+I believe these are better viewed side-by-side here rather than as separate UCRs, where the
+commonality would be lost from view. These are not complete descriptions of these services, just excerpts focusing on
+configuration, management, and status reporting.
+
+#### Trending Service
+The Trending or Data Historian Service computes simple statistical measures (eg, Minimum, Average, Maximum) for each 
+indicated Resource over a specified Interval and archives the results at the end of each interval. This is commonly
+done for 1/3 to 1/2 of the Resources produced by each south-bound device, though the user may choose to change the 
+configuration to Trend more or less of the Resources of any given Device, or choose not to Trend a Device at all. 
+
+Besides the overall configuration of this service (Interval size, depth of history to retain), it is necessary to configure which Resources from each Device of interest should be Trended, if any, and which statistical measures to use for each. This can be done with a Device which "extends" the original south-bound Device data, as described in
+the Use Case for Application Services Extending Device Data, [0003 Extending Device Data](https://github.com/edgexfoundry/edgex-docs/pull/800) *later (./0003-Extending-Device-Data.md)*; the Trending Service would add
+new per-Device Resources to do this configuration.
+
+#### Alarming Service
+The Alarming Service monitors Device Resources from Devices that belong to south-bound Device Services and 
+applies Alarm Rules to detect when a given Resource has entered or exited an Alarming state. 
+(Consider just the simple case of an Alarm Rule applied to one Resource at a time, which is the major Use Case 
+for Alarm Rules.) 
+
+There are just a few algorithms used for these simple Alarm Monitoring Rules, but there is per-Resource configuration 
+for different thresholds, hysteresis, delay, and severity levels; 
+sometimes these configurations are shared (eg, all 3 phases of voltage  measures are normally treated alike). Multiple threshold configurations may be grouped into
+an Alarm Group, so that a single Alarm lifecycle may flow through escalating severities (rather than disconnected 
+Alarm Events). The owner determines which Resources are monitored, with which algorithm, and at what levels.
+
+The design for this Alarming Service can also be done with Devices which "extend" each of the original south-bound 
+Devices, using Object-type Resources for the (moderately complex) Alarm Rules and Groups.
```

Comment:
Using Device Resource to store Alarm Rules and Groups?? this would case them to be on every instance of every device??

Reply:
Is this surprising? Two meters of the same model might set different alarm thresholds for the same measures (eg, current) depending on where in a building they are installed.

Reply:
OK, that make sense.